railties/lib/rails/commands/server/server_command.rb

```diff
@@ -44,17 +46,18 @@ def set_environment
     end
 
     def start
-      print_boot_information
       trap(:INT) { exit }
       create_tmp_directories
       setup_dev_caching
       log_to_stdout if options[:log_stdout]
 
       super
-    ensure
-      # The '-h' option calls exit before @options is set.
-      # If we call 'options' with it unset, we get double help banners.
-      puts "Exiting" unless @options && options[:daemonize]
+    end
+
+    def handler
+      server
+    rescue LoadError, NameError
+      raise MissingError
```

Comment:
Ugh. The last suggestion won't work because of `print_boot_information`. Maybe there's still something useful we can dig out of the suggestion though? 😊

Reply:
Kasper, I have tried something similar in the last round. I wasn't able to extract all the params out of `print_boot_information`, because `using` can be `nil`, or empty if we coerce it, but that still won't give us a server name to print.